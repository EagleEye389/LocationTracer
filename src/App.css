.app {
  margin:0;
  padding: 0;
  height: 80;
  font-size: 20px;
  z-index: 1;
}

.content{
  margin-left: 50px;
  margin-right: 50px;
  height:100vh;
  box-shadow: 5px 5px 5px gray;
  position: relative;
}

.pathSelectArea{
  min-height: 250px;
  background-color: gray;
  padding: 10px;
  z-index: 999;
  display: inline;
  position: absolute;
  top:0;
  left:0;  
  right: 65%;
  opacity: 1.0;
  bottom:90%;  
  
}
.mapArea{
display: inline-block;
min-height: 100vh;
position: relative;
border: 1px solid gray;
width: 100%;
 background-color: green;
 z-index: 2;
 margin: 0px;
 padding: 0px;
}

button{
  text-align: center;
  padding: 5px;
  background-color: white;
  color:black;
  margin-right: 50px;
}

.topBarTime{
  background-color: gray;
  color: white;
  width: auto;
  display: inline-block;
  z-index: 9999;
  position: absolute;
  right: 50%;
  height: 30px;
  padding: 5px;
}

.topBarDistance{
  background-color: gray;
  color: white;
  width: auto;
  display: inline-block;
  z-index: 9999;
  position: absolute;
  right: 0;
  height: 30px;
  padding: 5px;
}

.error{
  font-size: 13px;
  color: red;
  font-weight:bold;
  font-family: 'Times New Roman', Times, serif;
}