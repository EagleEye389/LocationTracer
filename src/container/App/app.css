.app {
  margin:0;
  padding: 0;
  font-size: 17px;
}

.content{
  margin-left: 20px;
  margin-right: 20px;
  height:95vh;
  box-shadow: 5px 5px 5px 5px gray;
  position: relative;
  display: flex;
  border-radius: 20px;
}

@media screen and (max-width:600px){
  .content{
    margin:0px;
    }
}