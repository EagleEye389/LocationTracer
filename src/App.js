import React,{Component} from 'react';
import './App.css';
import Axiosinstance from './Axios/axiosSetup';
import InputControl from './components/AutoInput/InputControl';
import Info from './components/Information/Information';
import ButtonControl from './components/button/buttonControl';
import normalizedLocation from './Helper/utiltiy';
import Map from './components/map/googleMap';
import {startLabel ,dropLabel, dropPlaceholder,
  startPlaceholder,timeLabel,distanceLabel} from './Helper/constant'

class App extends Component{
  constructor(props){
    super(props);
    this.state = {
        source :"",
        destination:"",
        distance:"",
        time:"",
        status:"",
        errorMsg:"",
        path:[],
        toggle:false,
        submitLabel:'Submit',
        valueEmpty:false
    }
  }
  
  // set source
  checkoutSource=(pro)=>{
    this.setState({source:pro.description})
  }
  
  //set destination
  checkoutDestination=(pro)=>{
     this.setState({destination:pro.description})
  }
  
// Reset tiggering and reset the value for next trigger
   resetValue=()=>{
     this.setState({
       valueEmpty:false
     })
   }

// On button reset button reset the data
  handleResetHandler=()=>{
    this.setState({
      destination:"",
      source:"",
      time:"",
      distance:"",
      path:[],
      toggle:!this.state.toggle,
      errorMsg:"",
      status:"",
      valueEmpty:true,
      submitLabel:'Submit'

    })
  }

  // Set the query to mock api
  handleSubmission=()=>{
  
  if(this.state.source && this.state.destination){  
    // get the token
    Axiosinstance.post('/route' ,{
                                  "origin": this.state.source,
                                  "destination": this.state.destination
                                }
    ).then((result)=>{
    let token = result.data.token;
     
    // If token is present the get the path and other data.
     const check =()=>{

     
           Axiosinstance.get(`/route/${token}`).then((ch)=>
                    {
                      console.log(ch.data.status);
                       

                   switch(ch.data.status){
                    case "success" : this.setState({
                          time : ch.data.total_time,
                          status: ch.data.status,
                          path : normalizedLocation(ch.data.path),
                          distance: ch.data.total_distance,
                          errorMsg:"",
                          submitLabel:'Re-Submit'
                      });break;
                      case "failure" : this.setState({
                        errorMsg: ch.data.error,
                        status:ch.data.status
                      });break;
                      case "in progress": 
                          this.setState({
                            errorMsg:"In progress,checking ...",
                            status:ch.data.status
                          });check();break;

                      default:break;
                    }
                     
                        
                      }).catch(err=>{console.log(err);
                        this.setState({
                          errorMsg:err.message,
                          status:"failure"
                        })
                      });
                    }
        // call the rescursive function in case "in progress"
         check();

    }).catch((error)=>{
      console.log(error);
      alert("Internal server error , please try again !!")
    })    
  }
  else
  {
    alert("Source or destination is/are missing.")
  }
  }
   
  render(){

    return (
      <div className="app">
       <div className="content">  
         <Info 
           classStyle ='topBarTime' 
           label ={timeLabel}
           value={this.state.time} />
         <Info 
          classStyle ='topBarDistance' 
          label ={distanceLabel} 
          value={this.state.distance} />

            <div className="pathSelectArea">
             <InputControl
              reset={this.resetValue}
              isReset={this.state.valueEmpty}
              label={startLabel}  
              placeholder={startPlaceholder}
              selection={ this.checkoutSource}
              />

            <InputControl
              reset ={this.resetValue}
              isReset ={this.state.valueEmpty}
              label={dropLabel}  
              placeholder={dropPlaceholder }
              selection={ this.checkoutDestination}
              />                     
            
            <br/><br/>
            {this.state.errorMsg && <span className="error">{this.state.errorMsg}</span>}
            <br/>
           < ButtonControl styler="button" label={this.state.submitLabel} handler={this.handleSubmission} />
           < ButtonControl styler="button" label="Reset" handler={this.handleResetHandler} />
           <br/>
           
          </div>
          <div className="mapArea">
           <Map 
            toggle={this.state.toggle}
             path={this.state.path} 
             />
          </div>
      </div>
       </div>   

  
    )
}
}

export default App;
