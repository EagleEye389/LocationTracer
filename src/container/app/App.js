import React, { Component } from "react";

import Selection from "../selection/SelectionContainer";
import Map from "../../components/maps/Maps";
import Loader from "../../components/loader/Loader";

import "./app.css";

/**
 * @type {Component}
 * @name APP
 * @description Main container.
 */
class App extends Component {
  state = {
    path: [],
    isLoading: false
  };

  /**
   * @name updateMap
   * @description This method to update path to draw polyline between source and destination
   * @param {{path}} Array of latitude and logtitude.
   */

  updateMap = path => {
    this.setState({ path: path });
  };

  /**
   * @name resetMap
   * @description This method to reset path to remove polyline from map.
   */
  resetMap = () => {
    this.setState({ path: [] });
  };

  /**
   * @name changeLoader
   * @description This method to reset loader.
   */
  changeLoader = isLoading => {
    this.setState({ isLoading });
  };

  render() {
    return (
      <div className="app">
        <div className="container">
          <div className="content">
            <Loader isLoading={this.state.isLoading} />
            <div className="row">
              <div className="col-xs-12 col-md-3 col-sm-12 col-lg-3">
                <Selection
                  updatePath={this.updateMap}
                  resetMap={this.resetMap}
                  changeLoader={this.changeLoader}
                />
              </div>
              <div className="col-xs-12 col-md-9 col-sm-12 col-lg-9">
                <Map path={this.state.path} />
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default App;
