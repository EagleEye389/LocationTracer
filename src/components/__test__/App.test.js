import React from "react";
import { shallow } from "enzyme";

import App from "../../container/App/App";

let component;
describe("Test component is rendering", () => {
  beforeEach(() => {
    component = shallow(<App />);
  });

  it("Should render basic control", () => {
    expect(component.find("div[class='content']")).toBeTruthy();
  });

  it("Should check app snapshot", () => {
    expect(component).toMatchSnapshot();
  });
});
