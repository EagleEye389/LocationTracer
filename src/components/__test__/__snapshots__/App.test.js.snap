// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component is rendering Should check app snapshot 1`] = `ShallowWrapper {}`;
